neuro_extras/data/archive.py

```diff
@@ -155,8 +156,16 @@ async def compress(self, source: Path, destination: Path) -> Path:
                 f"Supported types: "
                 f"{ArchiveType.get_extensions_for_type(ArchiveType.GZ)}"
             )
-        args = ["-r", str(destination), str(source)]
+        if source.is_dir():
+            raise ValueError(
+                "gzip does not support folder compression, use tar instead"
+            )
```

Comment:
It would still leave the files archived inplace (e.g. `folder/file.txt` -> `folder/file.txt.gz`), which is not intuitive nor trivial behaviour, as it does not allow to compress a folder into a single `.gz` archive. Using `.tar.gz` resolves this issue.

Reply:
perhaps, we should change the message to something like `Use .tar.gz extension instead`